cli/src/build-settings.ts

```diff
@@ -13,35 +14,46 @@ function isStringDict(x: JSON.Object): x is Dict<string | null> {
 
 export default class BuildSettings {
   private rustc: string;
+  private nodeVersion: string | null;
   private env: Dict<string | null>;
 
-  constructor(rustc: string, env: Dict<string | null>) {
+  constructor(rustc: string, nodeVersion: string | null, env: Dict<string | null>) {
     this.rustc = rustc;
+    this.nodeVersion = nodeVersion
     this.env = env;
   }
 
   match(other: BuildSettings) {
+    if (other.nodeVersion !== this.nodeVersion) return false;
     return Object.keys(this.env).every(key => {
       return (!this.env[key] && !other.env[key]) ||
              (this.env[key] === other.env[key]);
     });
   }
 
-  static current(toolchain: rust.Toolchain) {
-    const spawnResult = rust.spawnSync("rustc", ["--version"], toolchain);
+  static getNodeVersion(): string {
+    const nodeVersionResult = child_process.spawnSync("node", ["--version"]);
```

Comment:
I'm personally not in favor of the `const`-everywhere style, but mostly I think we should be consistent. Right now, we've only use `const` for constants -- i.e., global variables bound to immutable values. Would you be OK with just using `let` to be consistent with the house style?

Reply:
I'd love to! Whatever it takes to keep the style consistent. My bad for not catching this in advance.